aiidalab_widgets_base/structures.py

```diff
@@ -1,88 +1,138 @@
 from __future__ import print_function
 from __future__ import absolute_import
 
+import os
 import ase.io
 import ipywidgets as ipw
 from fileupload import FileUploadWidget
 import tempfile
 import nglview
 from six.moves import zip
+from traitlets import Bool
 
 
 class StructureUploadWidget(ipw.VBox):
-
+    has_structure = Bool(False)
     DATA_FORMATS = ('StructureData', 'CifData')
-
-    def __init__(self, text="Upload Structure", node_class=None, **kwargs):
+    def __init__(self, text="Upload Structure", storable=True, node_class=None, examples=[], **kwargs):
         """ Upload a structure and store it in AiiDA database.
 
         :param text: Text to display before upload button
         :type text: str
+        :param storable: Whether to provide Store button (together with Store format)
+        :type storable: bool
         :param node_class: AiiDA node class for storing the structure.
             Possible values: 'StructureData', 'CifData' or None (let the user decide).
             Note: If your workflows require a specific node class, better fix it here.
+        :param examples: list of paths to the example structures
+        :type examples: list
 
         """
-
         self.file_upload = FileUploadWidget(text)
+        self.select_example = ipw.Dropdown(
```

Comment:
can this be moved down under "if examples"?

Reply:
After discussion we agreed that I will keep it here for the moment